diff --git a/app/controllers/clipboard_controller.rb b/app/controllers/clipboard_controller.rb
index 0eff17b..1c8183b 100644
--- a/app/controllers/clipboard_controller.rb
+++ b/app/controllers/clipboard_controller.rb
@@ -3,7 +3,7 @@ class ClipboardController < ApplicationController
   before_action :require_existing_target_folder, :only => [:copy, :move]
   before_action :require_target_is_not_child, :only => :move
   before_action :require_create_permission, :only => [:copy, :move]
-  before_action :require_read_permission, :only => :create
+  before_action :require_read_permission, :only => [:create, :copy, :move]
   before_action :require_delete_permission, :only => :move
 
   # @item is set in require_existing_item
